File: food-ordering-system/src/app/shared/models/Order.ts
import {DeliveryType, OrderStatus, PaymentMethod} from "./enums";
import {OrderItem, OrderItemRequest} from "./OrderItem";
import {RestaurantPure} from "./Restaurant";

export type OrderPure = {
  id: string;
  totalPrice: number;

  status: OrderStatus;
  paymentMethod: PaymentMethod;
  deliveryType: DeliveryType;

  deliverTo?: string;

  restaurantId: string;
  restaurant: RestaurantPure;

  appUserId: string;
}

export type Order = OrderPure & {
  orderItems?: OrderItem[];
}

export type OrderRequest = {
  id?: string;

  paymentMethod: PaymentMethod;
  deliveryType: DeliveryType;
  status: OrderStatus;
  deliverTo?: string;
  restaurantId: string;
  appUserId: string;

  orderItems?: OrderItemRequest[];
}

File: FoodOrderingApp/Public.DTO/v1/Order.cs
using Base.Contracts.Domain;
using Helpers.Base.Entities;

namespace Public.DTO.v1;

public class OrderPure : IDomainEntityId
{
    public Guid Id { get; set; }
    
    public decimal TotalPrice { get; set; }

    public OrderStatus Status { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public DeliveryType DeliveryType { get; set; }

    public string? DeliverTo { get; set; }

    public Guid RestaurantId { get; set; }
    public Guid AppUserId { get; set; }
}

public class Order : OrderPure 
{
    public ICollection<OrderItem>? OrderItems { get; set; }
}

public class OrderRequest
{
    public Guid? Id { get; set; }

    public PaymentMethod PaymentMethod { get; set; }
    public DeliveryType DeliveryType { get; set; }
    public OrderStatus Status { get; set; }
    public string? DeliverTo { get; set; }
    public Guid RestaurantId { get; set; }
    public Guid AppUserId { get; set; }
}

public class OrderWithItemsRequest : OrderRequest
{
    public ICollection<OrderItemRequest> OrderItems { get; set; } = new List<OrderItemRequest>();
}